src/Simplex/FileTransfer/Agent.hs

```diff
@@ -192,8 +194,9 @@ runXFTPRcvWorker c srv Worker {doWork} = do
               atomically $ assertAgentForeground c
               loop
             retryDone e = rcvWorkerInternalError c rcvFileId rcvFileEntityId (Just fileTmpPath) (show e)
-    downloadFileChunk :: RcvFileChunk -> RcvFileChunkReplica -> AM ()
-    downloadFileChunk RcvFileChunk {userId, rcvFileId, rcvFileEntityId, rcvChunkId, chunkNo, chunkSize, digest, fileTmpPath} replica = do
+    downloadFileChunk :: RcvFileChunk -> RcvFileChunkReplica -> Bool -> AM ()
+    downloadFileChunk RcvFileChunk {userId, rcvFileId, rcvFileEntityId, rcvChunkId, chunkNo, chunkSize, digest, fileTmpPath} replica onlyViaProxy = do
+      when onlyViaProxy $ whenM unknownNoProxy $ throwError $ XFTP XFTP.UNKNOWN_NO_PROXY
```

Comment:
```suggestion
      unlessM ((approvedRelays ||) <$> usesSocksProxy) $ throwError $ XFTP XFTP.UNKNOWN_NO_PROXY
```

also different error

Reply:
also this error should be AgentErrorType, not XFTPErrorType, so there may so other way to exit this function other than throwing error, so it's in AM monad.